### DISCUSSION TITLE
Huge memory spike, no idea why, logging / debug tips?

### DISCUSSION THREAD

**Discussion by olaj** [SixLabors/ImageSharp.Web#161] [Q&A]
I have just figured out that its something with ImageSharp that are causing some weird memory spikes on my Azure App Service.

Take a look at this baby 😨 

![image](https://user-images.githubusercontent.com/184458/116236628-96e12000-a75f-11eb-937d-5f351bb13633.png)

And that is a 32gb Linux App Service Plan. So well, its a big one.

How can i try to debug this? I guess it could be whatever and it's pretty high traffic and its hundreds of thousands images that is served.

Any tips on where to start?

Ps. I just bought a developer license :) I read something about support there if you want to look into it in more detail.

**Comment by olaj** [comment]
I see a lot of these btw in the log.


DEBUG: Parser: IN_PACKETDEBUG: Parser: DecryptingDEBUG: Parser: pktLen:684,padLen:9,remainLen:672DEBUG: Parser: IN_PACKETDATADEBUG: Parser: DecryptingDEBUG: Parser: HMAC size:20DEBUG: Parser: IN_PACKETDATAVERIFYDEBUG: Parser: Verifying MACDEBUG: Parser: IN_PACKETDATAVERIFY (Valid HMAC)DEBUG: Parser: IN_PACKETDATAAFTER, packet: CHANNEL_DATA (0)DEBUG: Parser: IN_PACKETBEFORE (expecting 16)
DEBUG: Parser: IN_PACKETDEBUG: Parser: DecryptingDEBUG: Parser: pktLen:684,padLen:9,remainLen:672DEBUG: Parser: IN_PACKETDATADEBUG: Parser: DecryptingDEBUG: Parser: HMAC size:20DEBUG: Parser: IN_PACKETDATAVERIFYDEBUG: Parser: Verifying MACDEBUG: Parser: IN_PACKETDATAVERIFY (Valid HMAC)DEBUG: Parser: IN_PACKETDATAAFTER, packet: CHANNEL_DATA (0)DEBUG: Parser: IN_PACKETBEFORE (expecting 16)
DEBUG: Parser: IN_PACKETDEBUG: Parser: DecryptingDEBUG: Parser: pktLen:876,padLen:19,remainLen:864DEBUG: Parser: IN_PACKETDATADEBUG: Parser: DecryptingDEBUG: Parser: HMAC size:20DEBUG: Parser: IN_PACKETDATAVERIFYDEBUG: Parser: Verifying MACDEBUG: Parser: IN_PACKETDATAVERIFY (Valid HMAC)DEBUG: Parser: IN_PACKETDATAAFTER, packet: CHANNEL_DATA (0)DEBUG: Parser: IN_PACKETBEFORE (expecting 16)

Seems like its one of them every one second or so.

**Reply by JimBobSquarePants** [reply]
I'm not sure what the error message is about  (_Seems Docker based following a quick google_) but do you have v1.0.2 installed?

https://github.com/SixLabors/ImageSharp.Web/releases/tag/v1.0.2

That fixed an issue with missing cache lookups that caused hanging which would cause unnecessary processing.

Do you have any information that specifically indicates that ImageSharp is the cause?

**Reply by JimBobSquarePants** [reply]
@olaj Looks like we've found a leak. I'll get a release out.

**Reply by JimBobSquarePants** [reply]
We've got a build out right now on our MyGet feed. 
https://www.myget.org/feed/sixlabors/package/nuget/SixLabors.ImageSharp.Web/1.0.3-alpha.0.4

I'll be looking at getting a release out on NuGet also.

**Reply by olaj** [reply]
Great!

I saw a new spike this morning, I have deployed a new version now. Let me know if i notice anything strange with this one!

**Reply by olaj** [reply]
I've actually got a new spike after i deployed the new version. Maybe this wasn't it then 🤔

**Reply by JimBobSquarePants** [reply]
You'll need to profile the application. Currently there's no information in this thread actually identifying ImageSharp.Web as a culprit for anything nefarious. The spikes are also being claimed back so the GC is certainly handling whatever the cause is.

**Reply by JimBobSquarePants** [reply]
New NuGet release available. https://www.nuget.org/packages/SixLabors.ImageSharp.Web/1.0.3
